### PR TITLE
docs(gitea): add section about user scopes, rewrite

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Add section about user scopes
- Rewrite

## Context

@viceice asked me to add a section about user scopes to the Gitea docs in:

- https://github.com/renovatebot/renovate/discussions/22670#discussioncomment-6143825

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/gitea/index.md [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
+Either the account should have full name and email address set to allow Renovate to establish Git identity, or the `gitAuthor` config option should be set.
+
+### If you do not set `username` or `gitAuthor`
+
+If you don't set `username` or `gitAuthor`, then you must use the user scope.
+User scopes need Gitea `v1.19` or higher.
+
+The user scope must have these permissions:
+
+- `repo`
```
Comment (viceice) [inline lib/modules/platform/gitea/index.md:26]: this is always needed, it's not a user scope 

https://blog.gitea.io/2023/03/gitea-1.19.0-is-released/#-restrict-access-token-permissions-20908httpsgithubcomgo-giteagiteaissues20908

File: lib/modules/platform/gitea/index.md [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
 
 - Set your PAT as a `token` in your `config.js` file
 - Set your PAT as an environment variable `RENOVATE_TOKEN`
 - Set your PAT when you run Renovate in the CLI with `--token=`
 
-Remember to set `platform=gitea` somewhere in your Renovate config file.
-Either the account should have full name and email address set to allow Renovate to estabilish Git identity, or the `gitAuthor` config option should be set.
+You must set `platform=gitea` in your Renovate config file.
+
+Either the account should have full name and email address set to allow Renovate to establish Git identity, or the `gitAuthor` config option should be set.
```
Comment (HonkingGoose) [inline lib/modules/platform/gitea/index.md:17]: > Either the account should have full name and email address set to allow Renovate to establish Git identity, or the `gitAuthor` config option should be set.

Reading the current docs it's not clear to me what's the best way to authenticate to the Gitea platform. It looks like there are three ways:

- full name and email address
- `gitAuthor` config option
- User scopes (need Gitea `v1.19`)

If possible, the maintainers should decide on _one method_, that we will then recommend.

Having one method makes it easier for me to restructure the document. Right now it's a confusing mess of config methods... 😄 

Let me know what you think maintainers. 😉

Comment (rarkins) [reply lib/modules/platform/gitea/index.md:17]: I think you need 1+3 combined

Comment (HonkingGoose) [reply lib/modules/platform/gitea/index.md:17]: I'm thinking of splitting things up into:

```markdown
## Authentication
### Basic requirements // put things that you always need here
### Full token // the "old way"
### User scopes // the new "recommended" way that relies on Gitea v1.19+
```

But I don't know what each heading should contain.

File: lib/modules/platform/gitea/index.md [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
 
-Renovate supports [Gitea](https://gitea.io) and the new fork [Forgejo](https://forgejo.org).
+Renovate supports [Gitea](https://gitea.io) and the [Forgejo](https://forgejo.org) fork.
 Forgejo exists since `v1.18.0` and is currently fully compatible to Gitea.
 
 ## Authentication
 
-First, [create a Personal Access Token](https://docs.gitea.io/en-us/api-usage/#authentication) for the bot account.
-Let Renovate use your PAT by doing _one_ of the following:
+First, [create a Personal Access Token (PAT)](https://docs.gitea.io/en-us/api-usage/#authentication) for the bot account.
```
Comment (rarkins) [inline lib/modules/platform/gitea/index.md:8]:
```suggestion
First, [create a Personal Access Token (PAT)](https://docs.gitea.io/en-us/api-usage/#authentication) for the bot account.
The bot account should have full name and email address configured.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**